apps/drec-api/src/pods/issuer/issuer.service.ts

```diff
@@ -112,6 +119,37 @@ export class IssuerService {
     return await this.issueCertificate(issuance);
   }
 
+  private async handleLeftoverReads(
+    group: DeviceGroup,
+    totalReadValueW: number,
+  ): Promise<number> {
+    // Logic
+    // 1. Get the accummulated read values from devices
+    // 2. Transform current value from watts to kw
+    // 3. Add any leftover value from group to the current total value
+    // 4. Separate all decimal values from the curent kw value and store it as leftover value to the device group
+    // 5. Return all the integer value from the current kw value (if any) and continue issuing the certificate
+
+    const totalReadValueKw = group.leftoverReads
+      ? totalReadValueW * 10 ** -3 + Number(group.leftoverReads)
+      : totalReadValueW * 10 ** -3;
+    const integerReadValueKw = Math.floor(totalReadValueKw);
+    const leftOverReadValueKw = this.roundDecimalNumber(
+      totalReadValueKw - integerReadValueKw,
+    );
+    await this.groupService.updateLeftOverRead(group.id, leftOverReadValueKw);
+
+    return integerReadValueKw;
+  }
+
+  private roundDecimalNumber(num: number): number {
```

Comment:
To be honest, I would create a function, that will take a number, and return it's integral and decimal part as a result.

Where `Math.floor(num)` is integral, and `num - Math.floor(num)` is decimal

Reply:
Also, to round  number to given precision, just use `Math.floor(number * precision) / precision`.

Of course working with floats in JS is cumbersome, cause there can be some off, but you can't do it better AFAIR.

Reply:
This approach will substract one digit at the end. If you try it with 1.15 you will get 0.14.
Try this: Math.floor(1.15 * 10 ** 2) / 10 ** 2

Reply:
That's because JS is retarded, and 

```
> 1.15 * 100
114.99999999999999
```

:p

Reply:
Exactly :))